src/main/java/smp/SMP.java

```diff
@@ -175,16 +174,16 @@ public static void performSMPAlgorithm(
 
         //final String finalMatchingGraph = myBipartiteGraph.getFinalMatchingGraphToString();
         //System.out.println(finalMatchingGraph);
-        System.out.println("Number of Rejections: " + numberOfRejections);
+        // System.out.println("Number of Rejections: " + numberOfRejections);
         // n^2 - 2n + 1
         final double kWorstCaseRejections = Math.round(
                 Math.pow(numberOfRowsAndColumns, 2) - (2 * numberOfRowsAndColumns) + 1);
         if((int)kWorstCaseRejections == numberOfRejections)
         {
-            System.out.println("This is a Worst Case");
+            // System.out.println("This is a Worst Case");
         }
 
-        myBipartiteGraph.writeFinalMatchingToFile(outputFile);
+        return myBipartiteGraph.getFinalMatchingGraphToString();
```

Comment:
Instead of writing to a file, which incredibly expensive, the method now returns a String with the solution.